plugins/solidigm/solidigm-log-page-dir.c

```diff
@@ -0,0 +1,292 @@
+
+	static struct log_description solidigm_log_descs[] = {
+		{ 0xC1, "Read Commands Latency Statistics" },
+		{ 0xC2, "Write Commands Latency Statistics" },
+		{ 0xC4, "Endurance Manager Statistics" },
+		{ 0xC5, "Temperature Statistics" },
+		{ 0xCA, "SMART Attributes" },
+	};
+	const int solidigm_log_size = sizeof(solidigm_log_descs) / sizeof(struct log_description);
+
+	static struct log_description all_vu_log_descs[] = {
+		{ 0xC0, "OCP SMART / Health Information Extended" },
+		{ 0xC1, "OCP Error Recovery or Read Commands Latency Statistics" },
+		{ 0xC2, "OCP Firmware Activation History or Write Commands Latency Statistics" },
+		{ 0xC3, "OCP Latency Monitor" },
+		{ 0xC4, "OCP Device Capabilities or Endurance Manager Statistics" },
+		{ 0xC5, "OCP Unsupported Requirements or Temperature Statistics" },
+		{ 0xCA, "SMART Attributes" },
+	};
+	const int all_vu_log_size = sizeof(all_vu_log_descs) / sizeof(struct log_description);
+
+	// Standard logs are less than 0xC0
+	if (lid < 0xC0)
+		return lid_desc_from_struct(lid, standard_log_size, standard_log_descs);
+	else if (uuid == OCP_UUID)
+		return lid_desc_from_struct(lid, ocp_log_size, ocp_log_descs);
+	// Otherwise these are Solidigm logs.
+	else if (uuid == SOLIDIGM_UUID)
+		return lid_desc_from_struct(lid, solidigm_log_size, solidigm_log_descs);
+	else if (uuid == NO_UUID)
+		return lid_desc_from_struct(lid, all_vu_log_size, all_vu_log_descs);
+
+	return "Unknown";
+}
+
+static void solidigm_supported_log_pages_print(const struct supported_log_pages *supported,
+					       const enum solidigm_uuid uuid)
+{
+	printf("Log Page Directory Log:\n");
+	printf("  Supported:\n");
+
+	for (int lid = 0; lid < sizeof(supported->supported) / sizeof(__u32); lid++) {
+		if (supported->supported[lid] == 0)
+			continue;
+
+		printf("    Log Page:\n");
+		printf("      %-16s0x%02x\n", "LID:", le32_to_cpu(lid));
+		printf("      %-16s%s\n", "Description:", lid_to_desc(lid, uuid));
+	}
+
+	printf("\n");
+}
+
+static void solidigm_supported_log_pages_json(const struct supported_log_pages *supported,
+					      const enum solidigm_uuid uuid)
+{
+	struct json_object *root = json_create_object();
+	struct json_object *supported_arry = json_create_array();
+
+	for (int lid = 0; lid < sizeof(supported->supported) / sizeof(__u32); lid++) {
+		if (supported->supported[lid] == 0)
+			continue;
+
+		struct json_object *supported_obj = json_create_object();
+
+		json_object_add_value_uint(supported_obj, "lid", le32_to_cpu(lid));
+		json_object_add_value_string(supported_obj, "description",
+					     lid_to_desc(lid, uuid));
+
+		json_array_add_value_object(supported_arry, supported_obj);
+	}
+
+	json_object_add_value_array(root, "supported", supported_arry);
+
+	json_print_object(root, NULL);
+	json_free_object(root);
+
+	printf("\n");
+}
+
+int solidigm_get_log_page_directory_log(int argc, char **argv, struct command *cmd,
+					struct plugin *plugin)
+{
+	const __u8 log_id = 0x00;
+	int uuid_index = 0;
```

Comment:
I sugest initialize uuid_index to a invalid value, then later check if it was changed through the command line.
If still invalid, run the report for all UUIDs listed be nvme_identify_uuid().

Reply:
Are you saying to iterate through Solidigm and OCP logs? Maybe we should just remove the UUID selection and iterate through all known UUIDs no matter what.

Reply:
Iterating over the whole UUID list would be even better, the code become generic and would work with any drive from any vendor. If you follow this path just make sure to list the VU pages from UUIDs we don't know about as unknown.

Reply:
Yeah that's what I gathered; Makes sense, but will likely need to add a UUID field as well. I'll see what I can do with regards to making things generic because ultimately the actual descriptions for VU logs will vary between vendor.